res/res_tonedetect.c

```diff
@@ -394,6 +396,16 @@ static int detect_callback(struct ast_audiohook *audiohook, struct ast_channel *
 			}
 			ast_debug(1, "TONE_DETECT just got a hit (#%d in this direction, waiting for %d total)\n", now, di->hitsrequired);
 			if (now >= di->hitsrequired) {
+				message = ast_channel_blob_create_from_cache(ast_channel_uniqueid(chan), ast_channel_tone_detect(), NULL);
+
+				if (!message) {
+					ast_log(LOG_ERROR, "unable to create ast_channel_tone_detect message");
```

Comment:
This should include the channel name.

"Unable to publish tone detected event for ARI on channel '%s'"

Reply:
Yeah, fixed :)